### PR TITLE
Check for ports after a '.'

### PR DESCRIPTION
**Pull request by darthscsi**
Tests coming, and checks for '.' at consumeif

### CODE REVIEW

**Comment by darthscsi** [comment]
Fixes #550

**Comment by darthscsi** [comment]
Need to factor out that lookup, since it's duplicate code.

File: lib/Dialect/FIRRTL/Import/FIRParser.cpp [hunk 1]
```diff
@@ -819,18 +823,44 @@ ParseResult FIRScopedParser::lookupSymbolEntry(SymbolValueEntry &result,
...
+}
 
-  if (!entry.is<Value>())
-    return emitError(loc, "bundle value '" + name.str() +
-                              "' should only be used from subfield"),
-           failure();
+ParseResult FIRScopedParser::resolveSymbolEntry(Value &result,
+                                                SymbolValueEntry &entry,
+                                                StringRef fieldName, SMLoc loc,
+                                                bool fatal) {
```
Comment (lattner) [inline lib/Dialect/FIRRTL/Import/FIRParser.cpp:841]: I think fatal is always true here, you can drop it to simplify some code.

File: lib/Dialect/FIRRTL/Import/FIRParser.cpp [hunk 2]
```diff
@@ -1348,10 +1357,26 @@ FIRStmtParser::parseExpWithLeadingKeyword(StringRef keyword,
...
-      parseOptionalExpPostscript(lhs, subOps))
+  SymbolValueEntry symtabEntry;
+
+  if (lookupSymbolEntry(symtabEntry, keyword, info.getFIRLoc()))
     return ParseResult(failure());
 
+  // If we have a '.', we might have a symbol or an expanded port.  If we
+  // resolve to a symbol, use that, otherwise check for expanded bundles of
+  // other ops
+  if (resolveSymbolEntry(lhs, symtabEntry, info.getFIRLoc(), false)) {
```
Comment (lattner) [inline lib/Dialect/FIRRTL/Import/FIRParser.cpp:1368]: This is reachable with <= and other tokens, I think you should explicitly check for FIRToken::period here, something like:

`if (getToken.is(FIRToken::period)...`

File: lib/Dialect/FIRRTL/Import/FIRParser.cpp [hunk 3]
```diff
@@ -1348,10 +1357,26 @@ FIRStmtParser::parseExpWithLeadingKeyword(StringRef keyword,
...
+    StringRef fieldName;
+    consumeToken(); // token is FIRToken::period
```
Comment (lattner) [inline lib/Dialect/FIRRTL/Import/FIRParser.cpp:1370]: `consumeToken(FIRToken::period);` will assert the token is the expected type.

**Review by lattner — APPROVED** [review]
great catch!

File: lib/Dialect/FIRRTL/Import/FIRParser.cpp (hunk 2 above)
Comment (darthscsi) [reply lib/Dialect/FIRRTL/Import/FIRParser.cpp:1368]: other tokens follow the same path as before.  I'll update the comment to be more clear.

**Pull request merged**